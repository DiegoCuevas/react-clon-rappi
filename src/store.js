import { createStore, applyMiddleware, compose } from "redux";
import reducer from "./reducer";
import thunk from "redux-thunk";
import logger from "redux-logger";
import offlineSync from "./midlewares/offline-sync";

const composer = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

if (localStorage.getItem("version") !== process.env.REACT_APP_VERSION) {
  localStorage.removeItem("state");
}
const preloadedState = JSON.parse(localStorage.getItem("state") || "{}");

const store = createStore(
  reducer,
  // preloadedState,
  // composer(applyMiddleware(thunk, logger, offlineSync))
  composer(applyMiddleware(thunk, logger))
);

export default store;
