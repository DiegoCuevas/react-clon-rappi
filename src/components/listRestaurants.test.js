import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";

import store from "../store";
import ListRestaurants from "./listRestaurants";

test("ListRestaurants component", () => {
  const { asFragment } = render(
    <Provider store={store}>
      <ListRestaurants />
    </Provider>
  );

  expect(asFragment()).toMatchSnapshot();
});
