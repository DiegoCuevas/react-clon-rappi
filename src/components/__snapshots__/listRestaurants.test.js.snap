// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListRestaurants component 1`] = `
<DocumentFragment>
  .emotion-18 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.emotion-8 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #000;
  margin: 10px 5px;
  min-width: 30%;
}

@media (max-width:768px) {
  .emotion-8 {
    min-width: 100%;
  }
}

.emotion-7 {
  border-radius: .5em;
  box-sizing: border-box;
  border: 1px solid #fff;
  padding: 15px;
  width: 100%;
  -webkit-transition-property: all;
  transition-property: all;
  -webkit-transition-duration: 0.25s;
  transition-duration: 0.25s;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0;
}

.emotion-7:hover {
  cursor: pointer;
  border: 1px solid #e5edef;
  box-shadow: -1px 2px 5px 0px rgba(0,0,0,.12);
}

@media (max-width:768px) {
  .emotion-7 {
    border-radius: 0px;
    border-top: 1px solid #e5edef;
    border-buttom: 1px solid #e5edef;
  }
}

.emotion-1 {
  position: relative;
  width: 124px;
}

.emotion-0 {
  height: 100%;
  width: 100%;
  position: absolute;
  border-radius: .5em 0 0 .5em;
}

.emotion-6 {
  padding: 10px;
}

.emotion-2 {
  font-size: 16px;
  color: #333;
  font-weight: 400;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
  display: block;
}

.emotion-3 {
  font-size: 12px;
  color: #9faab7;
  margin: 0;
  min-height: 12px;
  text-overflow: ellipsis;
  display: block;
  white-space: nowrap;
  overflow: hidden;
}

.emotion-5 {
  font-size: 12px;
  line-height: 20px;
  color: #333;
  text-align: right;
}

.emotion-4 {
  color: #9faab7;
  display: inline-block;
  vertical-align: top;
  font-size: 5px;
  padding: 0 5px;
}

<section
    class="emotion-18"
  >
    <a
      class="emotion-8"
      href="/restaurant/0"
    >
      <div
        class="emotion-7"
      >
        <div
          class="emotion-1"
        >
          <img
            alt="restaurant"
            class="emotion-0"
            src="https://images.rappi.pe/restaurants_background/1839-1556640864.png?d=200x200&e=webp"
          />
        </div>
        <div
          class="emotion-6"
        >
          <span
            class="emotion-2"
          >
            Mc Donalds
          </span>
          <span
            class="emotion-3"
<<<<<<< HEAD
          />
=======
          >
            Desayuno - Internacional
          </span>
>>>>>>> b40b17746d4d5ce37cad312d283417344bb60be6
          <span
            class="emotion-5"
          >
            Envio S/. 6.65 
            <span
              class="emotion-4"
            >
              ●
            </span>
             50 min 
          </span>
        </div>
      </div>
    </a>
    <a
      class="emotion-8"
      href="/restaurant/1"
    >
      <div
        class="emotion-7"
      >
        <div
          class="emotion-1"
        >
          <img
            alt="restaurant"
            class="emotion-0"
            src="https://images.rappi.pe/restaurants_background/1839-1556640864.png?d=200x200&e=webp"
          />
        </div>
        <div
          class="emotion-6"
        >
          <span
            class="emotion-2"
          >
            Venomous
          </span>
          <span
            class="emotion-3"
<<<<<<< HEAD
          />
=======
          >
            Desayuno - Internacional
          </span>
>>>>>>> b40b17746d4d5ce37cad312d283417344bb60be6
          <span
            class="emotion-5"
          >
            Envio S/. 6.65 
            <span
              class="emotion-4"
            >
              ●
            </span>
             50 min 
          </span>
        </div>
      </div>
    </a>
  </section>
</DocumentFragment>
`;
